hashicorp/dev-portal
# refactor: tutorial eyebrowSlot

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-ksrefactor-tutorial-eyebrow-hashicorp.vercel.app/swingset/components/tutorialcard) 🔎
- [Asana task](https://app.asana.com/0/1202702246158644/1202822591039668) 🎟️

## 🗒️ What

This PR refactors the `TutorialCard` to render an `eyebrowSlot`, if provided. The default will remain as the tutorial read time. 

## 🤷 Why

This decouples the bookmark logic (and in the future progress) from the tutorial card presentation, making the base component more flexible. 

## 🧪 Testing

- Visit [the swingset page](https://dev-portal-git-ksrefactor-tutorial-eyebrow-hashicorp.vercel.app/swingset/components/tutorialcard)
- scroll down to check out the example with the bookmark slot. 
- also [visit a collection](https://dev-portal-git-ksrefactor-tutorial-eyebrow-hashicorp.vercel.app/terraform/tutorials/azure-get-started) view page to see the eyebrow slot in use with the connected bookmark button. 

## 💭 Anything else?

~~This will undo the change where all tutorial cards automatically rendered the bookmark icon. However, we can adjust that once we are doing the data integration to fetch the actual bookmark state.~~ Ended up updating all instances.
